tagstore/README.rst

```diff
@@ -34,17 +34,17 @@ Here is an example of using the Tagstore API::
 
     # Create a biology taxonomy:
     biology = tagstore.create_taxonomy("Biology", owner_id=None)
-    plant = tagstore.add_tag_to_taxonomy('plant', biology)
-    conifer = tagstore.add_tag_to_taxonomy('conifer', biology, parent_tag=plant)
-    cypress = tagstore.add_tag_to_taxonomy('cypress', biology, parent_tag=conifer)
-    pine = tagstore.add_tag_to_taxonomy('pine', biology, parent_tag=conifer)
-    aster = tagstore.add_tag_to_taxonomy('aster', biology, parent_tag=plant)
+    plant = biology.add_tag('plant')
+    conifer = biology.add_tag('conifer', parent_tag=plant)
+    cypress = biology.add_tag('cypress', parent_tag=conifer)
+    pine = biology.add_tag('pine', parent_tag=conifer)
+    aster = biology.add_tag('aster', parent_tag=plant)
 
     # Create a "sizes" taxonomy:
     sizes = tagstore.create_taxonomy("sizes", owner_id=None)
-    small = tagstore.add_tag_to_taxonomy('small', sizes)
-    med = tagstore.add_tag_to_taxonomy('med', sizes)
-    large = tagstore.add_tag_to_taxonomy('large', sizes)
+    small = sizes.add_tag('small')
+    med = sizes.add_tag('med')
+    large = sizes.add_tag('large')
```

Comment:
Since we're nitting, the [`set` examples below](https://github.com/open-craft/blockstore/pull/15/files#diff-68fd5bf025c3883a22f4b583aa68fd56R58) can be shortened to:
```diff
diff --git a/tagstore/README.rst b/tagstore/README.rst
index dc9f22a..267af95 100644
--- a/tagstore/README.rst
+++ b/tagstore/README.rst
@@ -55,29 +55,29 @@ Here is an example of using the Tagstore API::
     tagstore.add_tag_to(cypress, redwood)
 
     # Find all asters
-    set([e for e in tagstore.get_entities_tagged_with(aster)])
+    set(tagstore.get_entities_tagged_with(aster))
     # result: {dandelion}
 
     # plants
-    set([e for e in tagstore.get_entities_tagged_with(plant)])
+    set(tagstore.get_entities_tagged_with(plant))
     # result: {dandelion, redwood}
 
     # small plants
-    set([e for e in tagstore.get_entities_tagged_with_all({plant, small})])
+    set(tagstore.get_entities_tagged_with_all({plant, small}))
     # result: {dandelion}
 
     # plants, with no tag inheritance
-    set([e for e in tagstore.get_entities_tagged_with(plant, include_child_tags=False)])
+    set(tagstore.get_entities_tagged_with(plant, include_child_tags=False))
     # result: set()
 
     # conifers
-    set([e for e in tagstore.get_entities_tagged_with(conifer)])
+    set(tagstore.get_entities_tagged_with(conifer))
     # result: {redwood}
 
     # small things starting with "d"
-    set([e for e in tagstore.get_entities_tagged_with(
-        small, entity_types=['thing'], external_id_prefix='d'
-    )])
+    set(tagstore.get_entities_tagged_with(
+        plant, entity_types=['thing'], external_id_prefix='d'
+    ))
     # result: {dandelion}
```

(and changing the final `get_entities_tagged_with` to use the `plant` tag instead of the `small` tag better illustrates the `external_id_prefix`, since `{dandelion, redwood}` are tagged with `plant`).

Reply:
Worth also mentioning the list method too:
```python
# get tag hierarchy
list((tag.name, parent.name) for (tag, parent) in biology.list_tags_hierarchically())
# result: [('plant', None), ('aster', 'plant'), ('conifer', 'plant'), ('cypress', 'conifer'), ('pine', 'conifer')]
```

Reply:
I did one better (I think?):

```python
    # Print tag hierarchy tree:
    depths = {}
    for (tag, parent) in biology.list_tags_hierarchically():
        depths[tag] = depths[parent] + 1 if parent else 0
        print(("  " * depths[tag]) + tag.name)
    # The resulting hierarchy that gets printed out is:
    #   plant
    #     aster
    #     conifer
    #       cypress
    #       pine
```